frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1093,904 +1094,945 @@ function ReaderFooter:addToMainMenu(menu_items)
-                            self:onUpdateFooter()
-                            UIManager:setDirty(nil, "ui")
-                        end
-                    }
-                    UIManager:show(sort_item)
-                end,
-            },
-            getMinibarOption("all_at_once", self.updateFooterTextGenerator),
-            {
-                text = _("Hide empty items"),
-                help_text = _([[This will hide values like 0 or off.]]),
-                enabled_func = function()
-                    return self.settings.all_at_once == true
-                end,
-                checked_func = function()
-                    return self.settings.hide_empty_generators == true
-                end,
-                callback = function()
-                    self.settings.hide_empty_generators = not self.settings.hide_empty_generators
-                    self:refreshFooter(true, true)
-                end,
-            },
-            getMinibarOption("reclaim_height"),
-            {
-                text = _("Auto refresh"),
-                checked_func = function()
-                    return self.settings.auto_refresh_time == true
-                end,
-                callback = function()
-                    self.settings.auto_refresh_time = not self.settings.auto_refresh_time
-                    self:rescheduleFooterAutoRefreshIfNeeded()
-                end
-            },
-            {
-                text = _("Show footer separator"),
+                text = _("Show progress bar"),
                 checked_func = function()
-                    return self.settings.bottom_horizontal_separator == true
+                    return not self.settings.disable_progress_bar
                 end,
                 callback = function()
-                    self.settings.bottom_horizontal_separator = not self.settings.bottom_horizontal_separator
+                    self.settings.disable_progress_bar = not self.settings.disable_progress_bar
+                    if not self.settings.disable_progress_bar then
+                        self:setTocMarkers()
+                    end
+                    -- If the status bar is currently disabled, switch to an innocuous mode to display it
+                    if not self.view.footer_visible then
+                        self.mode = self.mode_list.page_progress
+                        self:applyFooterMode()
+                        G_reader_settings:saveSetting("reader_footer_mode", self.mode)
+                    end
                     self:refreshFooter(true, true)
                 end,
             },
             {
-                text = _("Lock status bar"),
-                checked_func = function()
-                    return self.settings.lock_tap == true
-                end,
-                callback = function()
-                    self.settings.lock_tap = not self.settings.lock_tap
+                text = _("Switch to chapter progress bar"),
```

Comment:
Dunno if `Switch to` is nice: it's a checkbox, other stuff witch checkbox don't say "Switch to show progress bar", just "Show progress bar". So, it probably should use the verb "Use" or something mentionning what is done after it is switched, not the action of "switching".